src/cascadia/TerminalApp/CommandPalette.cpp

```diff
@@ -181,23 +181,39 @@ namespace winrt::TerminalApp::implementation
         {
             // Action Mode: Dispatch the action of the selected command.
 
-            if (const auto selectedItem = _filteredActionsView().SelectedItem())
+            if (_currentMode == CommandPaletteMode::ActionMode)
             {
-                _dispatchCommand(selectedItem.try_as<TerminalApp::Command>());
+                if (const auto selectedItem = _filteredActionsView().SelectedItem())
+                {
+                    _dispatchCommand(selectedItem.try_as<TerminalApp::Command>());
+                }
+            }
+            else if (_currentMode == CommandPaletteMode::CommandlineMode)
+            {
+                _dispatchCommandline();
             }
 
             e.Handled(true);
         }
         else if (key == VirtualKey::Escape)
         {
-            // Action Mode: Dismiss the palette if the text is empty, otherwise clear the search string.
-            if (_searchBox().Text().empty())
+            if (_currentMode == CommandPaletteMode::ActionMode)
             {
-                _dismissPalette();
+                // Action Mode: Dismiss the palette if the text is empty,
+                // otherwise clear the search string.
+                if (_searchBox().Text().empty())
+                {
+                    _dismissPalette();
+                }
+                else
+                {
+                    _searchBox().Text(L"");
+                }
             }
-            else
+            else if (_currentMode == CommandPaletteMode::CommandlineMode)
             {
-                _searchBox().Text(L"");
+                // CommandlineMode Mode: always exit the palette immediately.
+                _dismissPalette();
```

Comment:
Shouldn't this be consistent with "ESC once to clear, twice to exit"? Long-term, switching modes is supposed to be which prefix we have, so they should line up right?

Reply:
Yea okay. I can work with that.